third_party/move/move-model/src/ty.rs

```diff
@@ -1718,7 +1718,7 @@ impl Substitution {
                     }
                 },
                 (Constraint::NoTuple, ty) => {
-                    if ty.is_tuple() {
+                    if ty.is_tuple() || ty.is_unit() {
```

Comment:
`is_tuple` seems to be used in another file in a few places. Might be worth checking if a similar issue could exist in those places?

Reply:
Thanks for pointing this out. I updated the implementation of `is_tuple` to include the unit type.

Reply:
There are 3 uses in `exp_builder.rs` that you may want to examine to decide if this treatment is appropriate.